src/effects/VST3/VST3UIValidator.cpp

```diff
@@ -181,10 +181,8 @@ void VST3UIValidator::OnClose()
 
 bool VST3UIValidator::UpdateUI()
 {
-   mWrapper.EndParameterEdit();
-   mWrapper.FetchSettings(mAccess.Get());
-   mWrapper.BeginParameterEdit(mAccess);
-
+   mAccess.ModifySettings([&](EffectSettings& settings) { mWrapper.FetchSettings(settings); });
```

Comment:
For all other effect families, ValidateUI() and UI event handlers use ModifySettings to start the inter-thread message to the running instance, but UpdateUI() is supposed to move data the other way -- take changes of instance state, and update the appearance of a dialog to reflect that; not to change instance state.

But here UpdateUI is doing something to the wrapper immediately in this thread, but ALSO starting the inter-thread communication that later causes ConsumeChanges().

Is all of that really needed to fix the bug?

Or, was ModifySettings done for inessential reasons, because now FetchSettings takes a non-const argument?

See the suggested extra commit I pushed and please tell me if it is a simplification that preserves the fixes, and squash it, or tell me it's definitely wrong and force-push the branch to remove it.